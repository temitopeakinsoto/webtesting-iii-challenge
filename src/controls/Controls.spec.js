// Test away
import React from 'react';
import * as rtl from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Controls from './Controls';

afterEach(rtl.cleanup);

let wrapper

beforeEach(() => {
  wrapper = rtl.render(<Controls />);
});

describe('Control Toggles On Click of Button', () => {
    it('clicking lock gate button doesn\'t open it', () => {
        expect(wrapper.queryByText(/close gate/i)).toBeInTheDocument();
        expect(wrapper.queryByText(/lock gate/i)).toBeDisabled();
        rtl.fireEvent.click(wrapper.queryByText(/close gate/i));
        expect(wrapper.queryByText(/lock gate/i).closest('button')).toHaveClass('toggle-btn');
        rtl.fireEvent.click(wrapper.queryByText(/lock gate/i));
        rtl.fireEvent.click(wrapper.queryByText(/close gate/i));
        expect(wrapper.queryByText(/close gate/i)).toHaveClass('toggle-btn');
    })
})