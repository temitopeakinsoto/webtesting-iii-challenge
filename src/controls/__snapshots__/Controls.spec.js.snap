// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Controls component display component matches the snapshot 1`] = `
<div>
  <div
    class="controls panel"
  >
    <button
      class="toggle-btn"
      disabled=""
    >
      Lock Gate
    </button>
    <button
      class="toggle-btn"
    >
      Close Gate
    </button>
  </div>
</div>
`;

exports[`Display component display component matches the snapshot 1`] = `
<div>
  <div
    class="controls panel"
  >
    <button
      class="toggle-btn"
      disabled=""
    >
      Lock Gate
    </button>
    <button
      class="toggle-btn"
    >
      Close Gate
    </button>
  </div>
</div>
`;
